### ISSUE TITLE
display assumptions used to create a symbol

### ISSUE THREAD

**Issue by smichr** [sympy/sympy#8042]
To reproduce a symbol with fidelity from srepr, the assumptions used to create the symbol must be shown. It would be best if the minimal set of _initial_ assumptions[1] would be captured in a canonical way and stored and then used when generating the srepr of the symbol.  I made an attempt at this at #8025 for reference.

Here's what I mean about finding the initial assumptions to store:

```
>>> given_assumptions_not_assumptions0=dict(nonzero=True, complex=True, zero=None)
>>> obj_assumptions0 = Dummy(**given_assumptions_not_assumptions0).assumptions0
>>> assumptions = dict([(k,v) for k,v in given_assumptions_not_assumptions0.items() if v is not None])
>>> if len(assumptions) == 1:
...    pass
... else:
...     for s in sorted(subsets(assumptions.items())):
...         if s and Dummy(**dict(s)).assumptions0 == obj_assumptions0:
...             assumptions = dict(s)
...             break
...
>>> assumptions  # this should be stored to use in srepr
{'nonzero': True}
```

**Comment by oscarbenjamin** [comment]
The stated problem doesn't seem to be an issue any more.

I think this was supposed to be closed by #8819.

**Comment by cbm755** [comment]
I agree!
